python/pyspark/rdd.py

```diff
@@ -2483,6 +2485,32 @@ def _is_barrier(self):
         """
         return self._jrdd.rdd().isBarrier()
 
+    def withResources(self, profile):
+        """
+        .. note:: Experimental
+
+        Specify a ResourceProfile to use when calculating this RDD. This is only supported on
+        certain cluster managers and currently requires dynamic allocation to be enabled.
+        It will result in new executors with the resources specified being acquired to
+        calculate the RDD.
+
+        .. versionadded:: 3.1.0
+        """
+        self.has_resourceProfile = True
+        self._jrdd.withResources(profile._jResourceProfile)
+        return self
+
+    def getResourceProfile(self):
```

Comment:
@tgravescs, can you clarify if we're going to make `getResourceProfile` and `withResources` as APIs or not first? I strongly think this is a bad idea to mix private and API declarations

https://github.com/apache/spark/blob/5d76b12e9b2ca0eb090c3c5145eee4cf78caba13/core/src/main/scala/org/apache/spark/rdd/RDD.scala#L1740-L1743

Reply:
I don't know what you mean, we are going with these apis. This is only in master branch, not in branch-3.0.  https://issues.apache.org/jira/browse/SPARK-29150 is to make the RDD ones public. At this point the core feature is complete so making the python ones and java ones public doesn't hurt anything and they will be only be in 3.1.0